packages/opa/tests/compile.test.ts

```diff
@@ -0,0 +1,188 @@
+import { describe, before, after, it } from "node:test";
+import assert from "node:assert";
+import { GenericContainer, StartedTestContainer, Wait } from "testcontainers";
+
+import { FilterCompileTargetsEnum, Filters, OPAClient } from "../src";
+import { OpaApiClientCore } from "../src/core.js";
+import { compileQueryWithPartialEvaluation } from "../src/funcs/compileQueryWithPartialEvaluation.js";
+import { CompileQueryWithPartialEvaluationAcceptEnum } from "../src/funcs/compileQueryWithPartialEvaluation.js";
+
+describe("compile-api", async () => {
+  const policies = {
+    filters: `package filters
+# METADATA
+# scope: document
+# custom:
+#   unknowns: [input.fruits]
+include if input.fruits.colour in input.fav_colours
+include if {
+  not input.fav_colours
+  endswith(input.fruits.name, "apple")
+}
+`,
+    filters_sans_metadata: `package filters_no_md
+include if input.fruits.colour in input.fav_colours
+include if {
+  not input.fav_colours
+  endswith(input.fruits.name, "apple")
+}
+`,
+  };
+
+  describe(
+    "with eopa",
+    {
+      // Skip these tests when there's no license env var (like dependabot PRs)
+      skip: !process.env["EOPA_LICENSE_KEY"],
+    },
+    () => {
+      let container: StartedTestContainer;
+      let serverURL: string;
+
+      after(async () => {
+        await container.stop();
+      });
+
+      before(async () => {
+        const opa = await prepareOPA(
+          "ghcr.io/styrainc/enterprise-opa:1.37.0",
```

Comment:
are we going to regret this pinning down the road?

Reply:
Eventually we should use latest here and in the other file, I think. But the mask stuff will hit edge earlier.